webproject/src/main/java/com/google/sps/servlets/OrderServlet.java

```diff
@@ -0,0 +1,90 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.Order;
+import com.google.sps.data.Room;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+@WebServlet("/order")
+public class OrderServlet extends HttpServlet {
+    private static Gson gson = new Gson();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Order newOrder = Order.newBuilder()
+                .setUser(""/*add get username here*/)
+                .setProduct(request.getParameter("product"))
+                .setQuantity(Integer.parseInt(request.getParameter("quantity")))
+                .setUnitPrice(Double.parseDouble(request.getParameter("unitPrice")))
+                .build();
+
+        String roomId = request.getParameter("roomId");
+        Room room = getRoom(roomId);
+        room.addOrder(newOrder);
+        String urlString = "https://summer20-sps-47.firebaseio.com/rooms/" + roomId + ".json";
```

Comment:
is there no API to update firebase? I believe there should be similar API like how you insert an Entity

Reply:
I would suggest to create a .save() method in Room class to do it. The method will update the firebase based on the roomId of the Room class

Reply:
if the roomId is empty, you can assume that it is an insertion to firebase

Reply:
I saw briefly but I didn't see one for Java, mostly for Kotlin or Javascript. But I created a class to handle it!

Reply:
since they are behind a class, I think we can leave it for now and come back later at other PR
https://firebase.google.com/docs/database/admin/save-data
do you think this is the doc for java?